src/netconf_confirmed_commit.c

```diff
@@ -0,0 +1,824 @@
+ */
+void
+ncc_set_persist(const char *value)
+{
+    char *new = NULL;
+
+    pthread_mutex_lock(&commit_ctx.lock);
+    free(commit_ctx.persist);
+    if (!value) {
+        commit_ctx.persist = NULL;
+        goto cleanup;
+    }
+
+    new = calloc(sizeof *new, strlen(value) + 1);
+    if (!new) {
+        ERR("Out of memory could not create new persist!");
+        goto cleanup;
+    }
+    strcpy(new, value);
+    commit_ctx.persist = new;
+
+cleanup:
+    pthread_mutex_unlock(&commit_ctx.lock);
+}
+
+/**
+ * @brief Safe getter for the timer commit_ctx_s member.
+ *
+ * @return timer
+ */
+timer_t
+ncc_get_timer(void)
+{
+    timer_t timer;
+
+    pthread_mutex_lock(&commit_ctx.lock);
+    timer = commit_ctx.timer;
+    pthread_mutex_unlock(&commit_ctx.lock);
+
+    return timer;
+}
+
+/**
+ * @brief Safe setter for the timer commit_ctx_s member.
+ *
+ * @param[in] value New value for timer.
+ */
+static void
+set_timer(timer_t timer)
+{
+    pthread_mutex_lock(&commit_ctx.lock);
+    commit_ctx.timer = timer;
+    pthread_mutex_unlock(&commit_ctx.lock);
+}
+
+/**
+ * @brief Trasform filename into module name.
+ *
+ * @param[in] filename Name of the file to extract module name out of.
+ *
+ * @return Module name.
+ */
+static char *
+get_module_name_from_filename(const char *filename)
+{
+    char *point = strstr(filename, ".json");
+    char *new = NULL;
+
+    if (!point) {
+        goto cleanup;
+    }
+
+    new = calloc(sizeof *new, point - filename);
+    if (!new) {
+        ERR("Out of memory, failed getting module name");
+        goto cleanup;
+    }
+
+    new = memcpy(new, filename, point - filename);
+
+cleanup:
+    return new;
+}
+
+/**
+ * @brief Rename a file if restore failed.
+ *
+ * @param[in] filename Name of the file to create new name from.
+ * @param[in] path Absolute path to the file.
+ */
+static void
+rename_failed_file(const char *filename, const char *path)
+{
+    char *new = NULL;
+
+    if (asprintf(&new, "%s/%s-%ld.failed", NP2SRV_DIR, filename, time(NULL)) == -1) {
+        ERR("Out of memory. Will not rename '%s' despite fail.", filename);
+        return;
+    }
+
+    if (rename(path, new)) {
+        ERR("Renaming '%s' failed.", filename);
+        goto cleanup;
+    }
+
+cleanup:
+    free(new);
+}
+
+/**
+ * @brief Read a backup file located on @p path and parse it
+ *
+ * @param[in] ctx Libyang context used for parsing.
+ * @param[in] path Path where the backup file is located.
+ * @param[out] node Parsed backup file into lyd_node.
+ *
+ * @return SR_ERR_LY When failed parsing.
+ * @return SR_ERR_OK When successful.
+ */
+static int
+get_running_backup(const struct ly_ctx *ctx, const char *path, struct lyd_node **node)
+{
+    int ret = SR_ERR_OK;
+
+    pthread_mutex_lock(&commit_ctx.lock);
+    if (lyd_parse_data_path(ctx, path, LYD_JSON, 0, LYD_VALIDATE_NO_STATE, node)) {
+        ERR("Failed parsing confirmed commit backup of running for file '%s'.", path);
```

Comment:
Would probably be useful to print the exact error, you can get it from the libyang context.